processor/metricstransformprocessor/README.md

```diff
@@ -56,6 +56,8 @@ transforms:
     new_name: <new_metric_name_inserted>
     # aggregation_type defines how combined data points will be aggregated; if action is combine, aggregation_type is required
     aggregation_type: {sum, mean, min, max}
+    # submatch_case specifies the case that should be used when adding label values based on regexp submatches when performing a combine action; leave blank to use the submatch value as is
+    submatch_case: {lower, upper}
```

Comment:
Yea that's a good point. It might be useful to have a map of named groups to config related to that group (for now just this replacement case). We probably also want more options than just upper and lower here. For now, I started with the simplest case and we can add more options in the future as needed. It shouldn't make too much difference to preserve backwards compatibility here if needed.